spyder/plugins/projects/plugin.py

```diff
@@ -225,6 +238,8 @@ def on_editor_teardown(self):
         widget.sig_project_loaded.disconnect(self._set_path_in_editor)
         widget.sig_project_closed.disconnect(self._unset_path_in_editor)
 
+        self._switcher.sig_open_file_requested.disconnect(editor.load)
```

Comment:
Following the comment above this could end up being something like `widget.sig_open_file_requested.disconnect(editor.load)`